### PR TITLE
Graphs support

### PR DESCRIPTION
**Pull request by dlasecki**

### CODE REVIEW

File: steps/graph.py [hunk 1]
```diff
@@ -33,14 +36,42 @@ def generate_random_regular_graph(
...
+        number_of_nodes: int, random_weights: bool = False, seed: Optional[int] = None
 ):
     graph = _generate_random_graph_erdos_renyi(
         number_of_nodes, 1.0, random_weights, seed
     )
     save_graph(graph, "graph.json")
 
 
+def generate_caveman_graph(
+        umber_of_cliques: int, size_of_cliques: int, random_weights: bool = False, seed: Optional[int] = None
```
Comment (mstechly) [inline steps/graph.py:48]:
```suggestion
        number_of_cliques: int, size_of_cliques: int, random_weights: bool = False, seed: Optional[int] = None
```

File: src/python/zquantum/core/graph_test.py [hunk 2]
```diff
@@ -117,63 +120,131 @@ def test_generate_random_regular_graph(self):
...
+    def test_generate_caveman_graph(self):
+        # Given
+        number_of_cliques = 2
+        size_of_cliques = 3
+
+        # When
+        graph = generate_caveman_graph(number_of_cliques, size_of_cliques)
+
+        # Then
+        assert type(graph) == nx.Graph
```
Comment (mstechly) [inline src/python/zquantum/core/graph_test.py:132]: I would either change it to two separate tests – one for checking the type and the other for checking if the graph was created properly.

Comment (mstechly) [reply src/python/zquantum/core/graph_test.py:132]: Same for other tests.

File: steps/graph.py [hunk 3]
```diff
@@ -9,10 +12,10 @@
 
 
 def generate_random_graph_erdos_renyi(
-    number_of_nodes: int,
-    edge_probability: float,
-    random_weights: bool = False,
-    seed: Optional[int] = None,
+        number_of_nodes: int,
```
Comment (mstechly) [inline steps/graph.py:15]: Looks to me like that's too much indentation.

**Review by mstechly — CHANGES_REQUESTED** [review]

**Comment by mstechly** [comment]
I've changed the target branch to `dev` and merged in latest changes.

File: src/python/zquantum/core/graph_test.py (hunk 2 above)
Comment (dlasecki) [reply src/python/zquantum/core/graph_test.py:132]: I got rid of this. It does not seem to bring in much value anyway.

File: steps/graph.py (hunk 3 above)
Comment (dlasecki) [reply steps/graph.py:15]: I think PEP wants an additional indentation when there is a line break.

Comment (mstechly) [reply steps/graph.py:15]: Fine 👍

**Review by mstechly — APPROVED** [review]

**Pull request merged**